src/NServiceBus.Core/Hosting/HostCreator.cs

```diff
@@ -59,11 +57,7 @@ public static async Task<IStartableEndpoint> CreateWithInternallyManagedContaine
 
             var hostingComponent = HostingComponent.Initialize(hostingConfiguration, serviceCollection, true);
 
-            var containerOptions = new ContainerOptions
-            {
-                EnableVariance = false
-            }.WithMicrosoftSettings();
-            var serviceProvider = serviceCollection.CreateLightInjectServiceProvider(containerOptions);
+            var serviceProvider = serviceCollection.BuildServiceProvider();
```

Comment:
There is an overload that takes a `ServiceProviderOptions` which has two options:

```csharp
BuildServiceProvider(new ServiceProviderOptions { ValidateOnBuild = true, ValidateScopes = true })
```

Both of those default to `false`, and it wasn't clear to me if we would want them to be true or not.

The documentation on those:
- [ValidateOnBuild](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.serviceprovideroptions.validateonbuild?view=dotnet-plat-ext-5.0#Microsoft_Extensions_DependencyInjection_ServiceProviderOptions_ValidateOnBuild)
- [ValidateScopes](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.serviceprovideroptions.validatescopes?view=dotnet-plat-ext-5.0#Microsoft_Extensions_DependencyInjection_ServiceProviderOptions_ValidateScopes)

Reply:
Going with the defaults sounds like a good move to me that would be the least surprising option for the users.